Python/ast.c

```diff
@@ -5591,6 +5592,28 @@ FstringParser_ConcatAndDel(FstringParser *state, PyObject *str)
     return 0;
 }
 
+/* Move *literal into the parser state object, thus transfering ownership.
+   Set *literal to NULL. If there's already a literal in the state object,
+   concatenate to it. */
+static int
+FstringParser_ConcatLiteral(FstringParser *state, PyObject **literal)
+{
+    if (!*literal) {
+        /* Do nothing. Just leave last_str alone (and possibly NULL). */
+    } else if (!state->last_str) {
+        /* Note that the literal can be zero length, if the input string is
```

Comment:
Sorry, but why this branch is needed? Why not use `FstringParser_ConcatAndDel` always?